src/models/hit-types/item-hit.ts

```diff
@@ -53,7 +68,7 @@ export class ItemHit {
    * Optional.
    */
   @Memoize() get collection_files_count(): NumberField | undefined {
-    return this.rawMetadata?.fields?.collection_files_count
+    return this.rawMetadata?.fields?.collection_files_count != null
```

Comment:
This (and similar elsewhere) are just to avoid returning undefined when the field actually exists with a falsey value.